common/Util.cpp

```diff
@@ -117,25 +117,33 @@ namespace Util
         {
             std::vector<char> v(length);
 
-            int len = 0;
-            /* use "/dev/[u]random" by default in case of
-            getentropy() or getrandom() fails,
-            or they don't available */
-            bool useDevRandom = true;
-#ifdef HAVE_GETENTROPY
-            len = getentropy(v.data(), length);
+            int len = -1;
+#if !HAVE_GETENTROPY && !HAVE_SYS_RANDOM_H
+#  error "getentropy or getrandom are required for succesful function";
```

Comment:
I think this is problematic for the iOS/Android cases, and seeing as the /dev/urandom fallback is restored, I guess drop this, or amend to be Linux only?

Reply:
Ah yes, I restored the fallback - we should not error out on iOS or Android I think - they're single-user and don't do the forkit stuff. We have a run-time error, but it'd be nicer to compile fail for Linux if it some ancient Linux I think.